core/modules/mysql/MySqlConfig.cc

```diff
@@ -55,23 +55,20 @@ MySqlConfig::MySqlConfig(std::string const& username,
                          std::string const& hostname,
                          unsigned int const port,
                          std::string const& socket,
-                         std::string const& dbName)
+                         std::string const& dbName,
+                         bool const checkValid)
     : username(username), password(password), hostname(hostname), port(port),
       socket(socket), dbName(dbName) {
 
-    if ((hostname.empty() or port == 0) and socket.empty()) {
-        LOGS(getLogger(), LOG_LVL_FATAL, "MySQL hostname:port and socket both undefined");
-        throw std::invalid_argument("MySQL hostname:port and socket both undefined" );
+    if (checkValid) {
+        checkValidity();
```

Comment:
ok I removed everything, we'll see later how to check configuration in a reasonable way.
Hope all MySQL config we'll be tested at program launch not to have a 'late' runtime error.